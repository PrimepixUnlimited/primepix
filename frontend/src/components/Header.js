import React from 'react';
import {Image, StyleSheet, View} from 'react-native';
import PropTypes from 'prop-types';
import {withNavigation} from 'react-navigation';
import {Header} from 'react-native-elements';

import Icon from './Icon';

import styles from '../constants/styles';

const HeaderComponent = ({
  navigation: {goBack, navigate},
  showBack,
  showTitle,
  showMenu,
  showProfile,
  title,
}) => {
  const handleBackPress = e => goBack();
  const handleMenuPress = e => navigate('menu');
  const handleProfilePress = e => navigate('profile');

  const Left = () => {
    return (
      <View style={s.leftContainer}>
        {showBack && (
          <Icon
            name="arrow-back"
            color="white"
            onPress={handleBackPress}
            underlayColor={styles.greyScale.black2}
          />
        )}

        {showMenu && (
          <Icon
            containerStyle={s.icon}
            name="menu"
            color="white"
            onPress={handleMenuPress}
            underlayColor={styles.greyScale.black2}
          />
        )}

        {/* {showTitle && (
          <Text h5 style={s.titleText}>
            {title}
          </Text>
        )} */}
      </View>
    );
  };

  const Middle = () =>
    showTitle && (
      <Image
        source={require('../assets/images/primepix-logo-darkmode.png')}
        style={s.logo}
      />
    );

  const Right = () =>
    showProfile && (
      <Icon
        containerStyle={s.icon}
        name="ios-person"
        color="white"
        type="ionicon"
        onPress={handleProfilePress}
        underlayColor={styles.greyScale.black2}
      />
    );

  return (
    <Header
      statusBarProps={{barStyle: 'light-content'}}
      containerStyle={s.container}>
      <Left />
      <Middle />
      <Right />
    </Header>
  );
};

HeaderComponent.propTypes = {
  showBack: PropTypes.bool,
  showProfile: PropTypes.bool,
  showMenu: PropTypes.bool,
  showTitle: PropTypes.bool,
  title: PropTypes.string,
};

HeaderComponent.defaultProps = {
  showBack: false,
  showProfile: false,
  showMenu: false,
  showTitle: true,
};

const s = StyleSheet.create({
  container: {
    backgroundColor: styles.greyScale.black2,
    borderBottomWidth: 0,
    shadowColor: styles.greyScale.black2,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 1,
    shadowRadius: 10,
    elevation: 23,
  },
  leftContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    width: 150,
  },
  icon: {
    backgroundColor: styles.greyScale.black2,
  },
  logo: {
    height: 35,
    resizeMode: 'cover',
    width: 70,
  },
  titleText: {
    color: 'white',
    marginLeft: 10,
  },
});

export default withNavigation(HeaderComponent);
