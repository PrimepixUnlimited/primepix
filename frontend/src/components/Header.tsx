import React, { FC } from 'react'
import { Image, StyleSheet, View } from 'react-native'
import { withNavigation } from 'react-navigation'
import { Header } from 'react-native-elements'
import {
  NavigationParams,
  NavigationRoute,
  NavigationScreenProp
} from 'react-navigation'

import ROUTES from '../navigation/_routes'

import Icon from './Icon'

import styles from '../constants/styles'

interface Props {
  navigation: NavigationScreenProp<NavigationRoute, NavigationParams>
  showBack?: boolean
  showTitle?: boolean
  showMenu?: boolean
  showProfile?: boolean
  title?: string
}

const HeaderComponent: FC<Props> = ({
  navigation: { goBack, navigate },
  showBack = false,
  showMenu = false,
  showProfile = false,
  showTitle = true,
  title
}) => {
  const handleBackPress = e => goBack(null)
  const handleMenuPress = e => navigate(ROUTES.Menu)
  const handleProfilePress = e => navigate(ROUTES.Profile)

  const Left = () => {
    return (
      <View style={s.leftContainer}>
        {showBack && (
          <Icon
            name="arrow-back"
            color="white"
            onPress={handleBackPress}
            underlayColor={styles.greyScale.black2}
          />
        )}

        {showMenu && (
          <Icon
            containerStyle={s.icon}
            name="menu"
            color="white"
            onPress={handleMenuPress}
            underlayColor={styles.greyScale.black2}
          />
        )}

        {/* {showTitle && (
          <Text h5 style={s.titleText}>
            {title}
          </Text>
        )} */}
      </View>
    )
  }

  const Middle = () =>
    showTitle && (
      <Image
        source={require('../assets/images/primepix-logo-darkmode.png')}
        style={s.logo}
      />
    )

  const Right = () =>
    showProfile && (
      <Icon
        containerStyle={s.icon}
        name="ios-person"
        color="white"
        type="ionicon"
        onPress={handleProfilePress}
        underlayColor={styles.greyScale.black2}
      />
    )

  return (
    <Header
      statusBarProps={{ barStyle: 'light-content' }}
      containerStyle={s.container}
    >
      <Left />
      <Middle />
      <Right />
    </Header>
  )
}

const s = StyleSheet.create({
  container: {
    backgroundColor: styles.greyScale.black2,
    borderBottomWidth: 0,
    shadowColor: styles.greyScale.black2,
    shadowOffset: {
      width: 0,
      height: 10
    },
    shadowOpacity: 1,
    shadowRadius: 10,
    elevation: 23
  },
  leftContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    width: 150
  },
  icon: {
    backgroundColor: styles.greyScale.black2
  },
  logo: {
    height: 35,
    resizeMode: 'cover',
    width: 70
  },
  titleText: {
    color: 'white',
    marginLeft: 10
  }
})

export default withNavigation(HeaderComponent)
