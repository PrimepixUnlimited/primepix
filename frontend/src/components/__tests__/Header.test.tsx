// components/__tests__/Hello.tsx
import "react-native";
import * as React from "react";
import * as renderer from "react-test-renderer";

import Header from "../Header";

describe("Header snapshot", () => {
  it("renders correctly", () => {
    const header = renderer
      .create(<Header showBack showTitle title="Header" />)
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
