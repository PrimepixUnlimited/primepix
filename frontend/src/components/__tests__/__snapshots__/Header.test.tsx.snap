// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#202124",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0,
      "elevation": 23,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingTop": 20,
      "shadowColor": "#202124",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 10,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": 150,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "resizeMode": "cover",
            "width": 70,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;
